### PR TITLE
RavenDB-14157 - Reduce the indexing tasks processing time

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14157

### CODE REVIEW

File: Raven.Database/Indexing/IndexingExecuter.cs [hunk 1]
```diff
@@ -258,17 +271,24 @@ private int ExecuteTask(HashSet<int> indexIds, HashSet<IComparable> alreadySeen)
...
-        private DatabaseTask GetApplicableTask(IStorageActionsAccessor actions, HashSet<IComparable> alreadySeen)
+        private DatabaseTask GetApplicableTask(IStorageActionsAccessor actions, ExecutedTasksInfo executedTasksInfo)
         {
             var disabledIndexIds = context.IndexStorage.GetDisabledIndexIds();
 
-            var removeFromIndexTasks = actions.Tasks.GetMergedTask<RemoveFromIndexTask>(
-                disabledIndexIds, context.IndexStorage.Indexes, alreadySeen);
-            if (removeFromIndexTasks != null)
-                return removeFromIndexTasks;
+            if (executedTasksInfo.SkipRemoveFromIndexTask == false)
```
Comment (arekpalinski) [inline Raven.Database/Indexing/IndexingExecuter.cs:277]: Why do we need this optimization? Can it be time consuming?

Comment (grisha-kotler) [reply Raven.Database/Indexing/IndexingExecuter.cs:277]: Yes. When we have disabled indexes, we are not going to execute the RemoveFromIndexTask.
If we have only TouchReferenceDocumentIfChangedTask we need to search for RemoveFromIndexTask although we didn't find any at the beginning.
Skipping tasks might be expensive if we have thousands of those.

Comment (arekpalinski) [reply Raven.Database/Indexing/IndexingExecuter.cs:277]: Okay. Thanks for explanation.

**Review by ayende — APPROVED** [review]

**Comment by ppekrol** [comment]
This is causing a lot of test failures. Closing due to inactivity. Please reopen when fixed.

**Pull request closed without merging**